tensorflow/python/ops/sparse_ops.py

```diff
@@ -1582,3 +1583,49 @@ def _SparseSparseMaximumMinimumShape(op):  # pylint: disable=invalid-name
   op.inputs[4].get_shape().assert_has_rank(1)  # b_values
   op.inputs[5].get_shape().assert_has_rank(1)  # b_shape
   return [tensor_shape.unknown_shape(2), tensor_shape.unknown_shape(1)]
+
+
+def sparse_transpose(sp_input, perm=None, name=None):
+  """Transposes a `SparseTensor`
+
+  The returned tensor's dimension i will correspond to the input dimension
+  `perm[i]`. If `perm` is not given, it is set to (n-1...0), where n is
+  the rank of the input tensor. Hence by default, this operation performs a
+  regular matrix transpose on 2-D input Tensors.
+
+  For example, if `sp_input` has shape `[4, 5]` and `indices` / `values`:
+
+      [0, 3]: b
+      [0, 1]: a
+      [3, 1]: d
+      [2, 0]: c
+
+  then the output will be a `SparseTensor` of shape `[5, 4]` and
+  `indices` / `values`:
+
+      [0, 2]: c
+      [1, 0]: a
+      [1, 3]: d
+      [3, 0]: b
+
+  Args:
+    sp_input: The input `SparseTensor`.
+    perm: A permutation of the dimensions of `sp_input`.
+    name: A name prefix for the returned tensors (optional)
+  Returns:
+    A transposed `SparseTensor`.
+
+  Raises:
+    TypeError: If `sp_input` is not a `SparseTensor`.
+  """
+  with ops.op_scope([sp_input], name, "SparseTranspose") as name:
```

Comment:
@maniteja123 You are right, let's keep it as a separate function for now. I think if you extend the tests to cover rank > 2 tensors, we should be good. Maybe it would be easier to generate random sparse tensors and use tf.sparse_tensor_to_dense(t) to convert it to dense, transpose the dense and compare with the "densified" version of the output from your function. See e.g. https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/sparse_ops_test.py#L721